processor/deltatorateprocessor/processor.go

```diff
@@ -41,10 +49,64 @@ func (dtrp *deltaToRateProcessor) Start(context.Context, component.Host) error {
 
 // processMetrics implements the ProcessMetricsFunc type.
 func (dtrp *deltaToRateProcessor) processMetrics(_ context.Context, md pdata.Metrics) (pdata.Metrics, error) {
+	resourceMetricsSlice := md.ResourceMetrics()
+
+	for i := 0; i < resourceMetricsSlice.Len(); i++ {
+		rm := resourceMetricsSlice.At(i)
+		ilms := rm.InstrumentationLibraryMetrics()
+		for i := 0; i < ilms.Len(); i++ {
+			ilm := ilms.At(i)
+			metricSlice := ilm.Metrics()
+			for j := 0; j < metricSlice.Len(); j++ {
+				metric := metricSlice.At(j)
+				_, ok := dtrp.metrics[metric.Name()]
+				if ok {
```

Comment:
Calculating rate also requires you to maintain the state. it looks you apply rate and only check the metric name are same which is not right. You should consider dimensions & resource values should all be matched as well.

Reply:
Hi Min, I want to explain a little bit here. I am not sure if I missed anything. This processor only coverts Delta to Rate. Our product requirement is to convert Cumulative Sum to Rate. We divided this into two part. 

1. Cumulative to Delta [We need the state here]. We update the usage and the startTimeStamp.
1. Delta to Rate [We don't need the state here]. We alreay have the usage and the time between startTimeStamp and current timeStamp. We can get the rate following `(usage) / (timeStamp - startTimeStamp)`. This was reviewed by Josh and Bogdan. Please check the (design doc)[https://docs.google.com/document/d/1NjeqzamzooqQP26hy5BWPvNsONLvdmV1bWqGwhMFoS4/edit#heading=h.br6h8f8h77c6]

Reply:
Got it. you directly calculate the rate against the selected metrics with delta value attached. Thanks.